### PR TITLE
Remove unused argument from `applications_page_check`

### PR DESCRIPTION
**Pull request by marksweb**
Addressing an issue from August 2014 here!!

## Description

Remove unused argument `current_page` from `applications_page_check`

* #3378

## Checklist

* [x] I have opened this pull request against ``develop``
* [ ] I have updated the **CHANGELOG.rst**
* [ ] I have added or modified the tests when changing logic
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by TiredFingers** [comment]
Signature changing always scary me. Using PyCharm I found 26 usages without any attemptions to use `current_page` argument. All checks passed. This change will not broke any. But I think we need somebody with expert level with django cms

**Comment by marksweb** [comment]
> Signature changing always scary me. Using PyCharm I found 26 usages without any attemptions to use `current_page` argument. All checks passed. This change will not broke any. But I think we need somebody with expert level with django cms

The issue this fixes was agreed by the core team in 2014 at which point it was assumed to be an argument leftover from some old functionality. So I've actioned it on the grounds it had been accepted by the people that know this system the best.

**Review by TiredFingers — APPROVED** [review]

**Review by victor-yunenko — APPROVED** [review]

**Comment by greengoaxe** [comment]
Hi @vinitkumar,
may you merge it?
Thanks,
greengoaxe

File: cms/appresolver.py
Comment (Aiky30) [inline cms/appresolver.py:30]: @marksweb How do you know that this is unused? This could be required for apphooks.

Comment (marksweb) [reply cms/appresolver.py:30]: @Aiky30 Based on the discussions on the associated issue (#3378) and a search of the project for use of this function.

The original issue is old, but has input from core members who agree that it's superfluous and likely stems from before the switch to treebeard.

**Comment by vinitkumar** [comment]
@greengoaxe Yes, I will merge it once the test passes.

**Pull request merged**
